media/src/components/textEditor/TextEditor.js

```diff
@@ -0,0 +1,259 @@
+//             editor={ClassicEditor}
+//             data={editorData}
+//             onReady={(editor) => {
+//               editor.editing.view.change((writer) => {
+//                 return writer.setStyle(
+//                   'height',
+//                   '200px',
+//                   editor.editing.view.document.getRoot()
+//                 );
+//               });
+//               // Set the editor instance to the ref
+//               editor.current = editor;
+//             }}
+//             onChange={handleEditorDataChange}
+//           />
+//           {imageUrl && <Image src={imageUrl} alt="Uploaded Image" />}
+//           <VStack mt={4}>
+//             <div dangerouslySetInnerHTML={{ __html: editorData }} />
+//           </VStack>
+//         </form>
+//       </Box>
+//     </Container>
+//   );
+// };
+
+// export default TextEditor;
+import { CKEditor } from "@ckeditor/ckeditor5-react";
+import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
+import { useEffect, useState } from "react";
+import { Box, Container } from "@chakra-ui/react";
+
+export default function TextEditor() {
+  const editorConfiguration = {
+    toolbar: {
+      items: [
+        "heading",
+        "|",
+        "bold",
+        "italic",
+        "link",
+        "bulletedList",
+        "numberedList",
+        "|",
+        "outdent",
+        "indent",
+        "|",
+        "imageUpload",
+        "blockQuote",
+        "insertTable",
+        "mediaEmbed",
+        "undo",
+        "redo",
+        "alignment",
+        "code",
+        "codeBlock",
+        "findAndReplace",
+        "fontColor",
+        "fontFamily",
+        "fontSize",
+        "fontBackgroundColor",
+        "highlight",
+        "horizontalAlignment",
+        "horizontalLine",
+        "htmlEmbed",
+        "imageInsert",
+      ],
+    },
+    language: "en",
+    image: {
+      toolbar: [
+        "imageTextAlternative",
+        "toggleImageCaption",
+        "imageStyle:inline",
+        "imageStyle:block",
+        "imageStyle:side",
+      ],
+    },
+    table: {
+      contentToolbar: ["tableColumn", "tableRow", "mergeTableCells"],
+    },
+  };
+
+  const [editorData, setEditorData] = useState(
+    "<p>Hello from CKEditor 5!</p>"
+  );
+
+  useEffect(() => {
+    // Initialize the CKEditor data
+    setEditorData("<p>Hello from CKEditor 5!</p>");
+  }, []);
+
+  // const handleEditorChange = (event: ChangeEvent, editor: any) => {
+  //   const data = editor.getData();
+  //   setEditorData(data);
+  // };
+
+  // const handleTextChange = (e: CKEDITOR.eventInfo) => {
+  //   console.log(e.editor.getData());
+  // };
+
+  return (
+    <>
+      <Container maxW="container.md">
+        <Box>
+          <h1>CKEditor Example</h1>
+          <form>
+            {/* <textarea
+              name="editor"
+              onInput={handleTextChange}
+              data-ck-editor="1"
+            ></textarea> */}
+          </form>
+          <CKEditor
```

Comment:
we use the tip tap library. you could have seen this by looking at the existing production code (outside repository) and there's no way you could have known that so that's on me! here's the [tiptap code](https://github.com/wayjake/Wallie.io/blob/master/src/Interface/TipTap.tsx).

Reply:
TipTap is more modern, declarative, and easier to style.